design/object-bucket/object-bucket.md

```diff
@@ -0,0 +1,127 @@
+# Rook-Ceph Object Bucket Controller
+
+## Proposed Feature
+
+Currently, the automation exists in the Rook-Ceph operator to support [Ceph Object Store](../object-store.md) and [Ceph Object Store User](https://github.com/rook/rook/issues/1583) custom resource definitions (CRDs).  
+
+This design closes the loop for Rook-Ceph object stores with a CephObjectBucket controller and CRD.  Users will request buckets via the Kubernetes API and have returned ConfigMaps with connection information for that bucket.  One ConfigMap will be created per CephObjectBucket.  ConfigMap names will be deterministic and based on the name of the CephObjectBucket.  This gives users the opportunity to define their Pod spec at the same time as the CephObjectBucket.  Additionally, the CephObjectBucket and Pod can safely be deployed at the same time thanks to built in synchronization in Kubernetes.
+
+Bucket deletion in the early stages of this controller design will not be addressed beyond cleaning up generated resources.  Deleting a CephObjectBucket will only cause deletion of the associated ConfigMap.  The controller will not delete the bucket from the object store. 
+
+## Dependencies
+
+This design assumes that cross-namespace support for CephObjectUsers, proposed in #2473, has been accepted into the project.  Specifically, the coupling of CephObjectStores to StorageClasses, which are then referenced by the CephObjectUser.
+
+## Object Bucket Walkthrough
+
+### Assumptions
+
+- The `storageClass` field has been added to the CephObjectStoreUser API
+- Rook-Ceph Object has been deployed in a Kubernetes Cluster ([_see:_ quick start](https://rook.github.io/docs/rook/master/object.html))
+- The `CephObjectStoreUser` has been created in the developer's namespace and the associated credentials are stored in a `Secret`
```

Comment:
what do you think of the possibility of generating a user on-demand for the bucket if there is no user specified? or do we want to be very explicit and keep that as a separate and required pre-requisite step?

Reply:
@copejon and I like that idea and it has been mentioned before. What is your opinion on phasing this to a later pr, or would this be the more common use case?

Reply:
@jbw976 @jeffvance Given that we're changing direction to a generalized provisioner rather than a ceph specific one, this would be a step back for us.  The reason being that we'd be coupled (again) to rook-ceph CRDs which is what we're trying to avoid.

Reply:
> ...we'd be coupled (again) to rook-ceph CRDs which is what we're trying to avoid.
+1